patterns/direct_requestor.py

```diff
@@ -0,0 +1,118 @@
+from solace.messaging.publisher.request_reply_message_publisher import RequestReplyMessagePublisher
+
+if platform.uname().system == 'Windows': os.environ["PYTHONUNBUFFERED"] = "1" # Disable stdout buffer 
+
+TOPIC_PREFIX = "solace/samples/python"
+
+name = ""
+while not name:
+    name = input("Enter your name: ")
+unique_name = name.replace(" ", "")
+
+# Inner classes for error handling
+class ServiceEventHandler(ReconnectionListener, ReconnectionAttemptListener, ServiceInterruptionListener):
+    def on_reconnected(self, e: ServiceEvent):
+        print("\non_reconnected")
+        print(f"Error cause: {e.get_cause()}")
+        print(f"Message: {e.get_message()}")
+    
+    def on_reconnecting(self, e: "ServiceEvent"):
+        print("\non_reconnecting")
+        print(f"Error cause: {e.get_cause()}")
+        print(f"Message: {e.get_message()}")
+
+    def on_service_interrupted(self, e: "ServiceEvent"):
+        print("\non_service_interrupted")
+        print(f"Error cause: {e.get_cause()}")
+        print(f"Message: {e.get_message()}")
+
+
+# Broker Config. Note: Could pass other properties Look into
+broker_props = {
+    "solace.messaging.transport.host": os.environ.get('SOLACE_HOST') or "tcp://localhost:55555,tcp://localhost:55554",
+    "solace.messaging.service.vpn-name": os.environ.get('SOLACE_VPN') or "default",
+    "solace.messaging.authentication.scheme.basic.username": os.environ.get('SOLACE_USERNAME') or "default",
+    "solace.messaging.authentication.scheme.basic.password": os.environ.get('SOLACE_PASSWORD') or ""
+    }
+
+# Build A messaging service with a reconnection strategy of 20 retries over an interval of 3 seconds
+# Note: The reconnections strategy could also be configured using the broker properties object
+messaging_service = MessagingService.builder().from_properties(broker_props)\
+                    .with_reconnection_retry_strategy(RetryStrategy.parametrized_retry(20,3))\
+                    .build()
+
+# Blocking connect thread
+messaging_service.connect()
+print(f'\nMessaging Service connected? {messaging_service.is_connected}')
+
+# Event Handling for the messaging service
+service_handler = ServiceEventHandler()
+messaging_service.add_reconnection_listener(service_handler)
+messaging_service.add_reconnection_attempt_listener(service_handler)
+messaging_service.add_service_interruption_listener(service_handler)
+
+# Create a direct message requestor and register the error handler
+direct_requestor: RequestReplyMessagePublisher = messaging_service.request_reply() \
+                                                                .create_request_reply_message_publisher_builder() \
+                                                                .build()
+
+# Blocking Start thread
+direct_requestor.start()
+print(f'\nDirect Requestor ready? {direct_requestor.is_ready()}')
+
+# Prepare outbound message payload and body
+message_body = "This is request message from " + f'{name}'
```

Comment:
`message_body = f'This is request message from {name} '

Reply:
Done!